demo/admin/src/products/future/generated/ProductForm.tsx

```diff
@@ -106,14 +107,14 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
                                 depth: String(data.product.dimensions.depth),
                             }
                           : undefined,
-                      availableSince: data.product.availableSince ? new Date(data.product.availableSince) : undefined,
+                      availableSince: hideFields?.availableSince && data.product.availableSince ? new Date(data.product.availableSince) : undefined,
```

Comment:
I believe this should be `!hideFields?.availableSince && data.product.availableSince`, otherwise the field will never be initialized.